docs/types.rst

```diff
@@ -352,6 +352,24 @@ check the value ranges at runtime and a failure causes an exception.  Enums need
         }
     }
 
+.. index:: ! contract type, ! type; contract
+
+.. _contract_types:
+
+Contracts
+---------
+
+Contracts themselves are their own individual type, as defined by their contract, that you can
```

Comment:
@chriseth any wording suggestion? I think this is a bit confusing right now :)

Reply:
> Contracts types are their own type, as defined by their smart contract declaration.

> A contract type is defined by its smart contract definition.

I can keep going with ideas here…

Reply:
I think the second one is reasonable. Maybe put in a link to the contract section which explains what kind of members a contract can have?